### PR TITLE
Fixed an issue where it was impossible to upload any icons in page settings.

### LINKED ISSUE #3863: Cannot upload svg files for page icons
## Description of bug
Pages support a small and a large icon, however there is no way using the UI to upload an svg file.

## Steps to reproduce
List the precise steps to reproduce the bug:
1. Go to any page
2. Try to upload an svg image for the page icon

## Current behavior
Nothing happens, the file appears to be uploading but then it does not show in the UI or in the database, no error is present either in the browser console.

## Expected behavior
Should upload and show up right away in the UI and be saved in the database.

## Additional context
- Setting the icon manually in the database does make it appear both in the pages UI and where it renders in menu systems that support it.
- .svg is in the allowed files extensions everywhere and there is no issue uploading the same file using the file manager.
- uploading first the file in the file manager and then trying to select it in the pages UI also does not work, it does not show svg files in the selection list, jpg and gif do show.

## Affected version

* [?] 10.00.00 alpha build (in development)
* [x] 09.06.02 release candidate
* [x] 09.06.01 latest supported release (Well there where more problems there with this feature)

### PR DESCRIPTION
**Pull request by valadas**
- Resolves about 200 Stylecop warnings
- Fixes an issue where filetypes where detected wrong from mimeTypes (for instance "application/svg+xml" was loogink for files named *.svg+xml" it now properly looks for *.svg files instead).
- Made the validationCode required in the FileUpload common component since it does not work without that prop.
- Fixes #3863 in a similar fashion as in #3769 but for the page icons.

### CODE REVIEW

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs [hunk 1]
```diff
@@ -534,117 +651,137 @@ public HttpResponseMessage SavePageAsTemplate(PageTemplate pageTemplate)
...
-                this._tabController.ConvertTabToNeutralLanguage(this.PortalId, pageId, defaultLocale.Code, true);
+                var defaultLocale = this.localeController.GetDefaultLocale(this.PortalId);
+                this.tabController.ConvertTabToNeutralLanguage(this.PortalId, pageId, defaultLocale.Code, true);
                 return this.Request.CreateResponse(HttpStatusCode.OK, new { Success = true });
             }
             catch (Exception ex)
             {
-                Logger.Error(ex);
-                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.ToString());
+                var errorMessage = "An unexpected error occured while trying to make this page neureal, please consult the logs for more details.";
```
Comment (david-poindexter) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs:683]:
```suggestion
                var errorMessage = "An unexpected error occurred while trying to make this page neutral, please consult the logs for more details.";
```

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs [hunk 2]
```diff
@@ -534,117 +651,137 @@ public HttpResponseMessage SavePageAsTemplate(PageTemplate pageTemplate)
...
-                this._tabController.ClearCache(this.PortalId);
+                this.tabController.AddMissingLanguages(this.PortalId, pageId);
+                this.tabController.ClearCache(this.PortalId);
                 return this.Request.CreateResponse(HttpStatusCode.OK, new { Success = true });
             }
             catch (Exception ex)
             {
-                Logger.Error(ex);
-                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.ToString());
+                var errorMessage = "An unexpected error occured while trying to add missing languages to this page, consult the logs for more details.";
```
Comment (david-poindexter) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs:751]:
```suggestion
                var errorMessage = "An unexpected error occurred while trying to add missing languages to this page, consult the logs for more details.";
```

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs [hunk 3]
```diff
@@ -656,53 +793,61 @@ public HttpResponseMessage NotifyTranslators(TranslatorsComment comment)
             }
             catch (Exception ex)
             {
-                Logger.Error(ex);
-                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.ToString());
+                var errorMessage = "An unexpected error occured while trying to notify the translators, please consult the logs for more details.";
```
Comment (david-poindexter) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs:796]:
```suggestion
                var errorMessage = "An unexpected error occurred while trying to notify the translators, please consult the logs for more details.";
```

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs [hunk 4]
```diff
@@ -656,53 +793,61 @@ public HttpResponseMessage NotifyTranslators(TranslatorsComment comment)
...
                     pages = this.GetNonLocalizedPages(pageId);
                 }
+
                 return this.Request.CreateResponse(HttpStatusCode.OK, pages);
             }
             catch (Exception ex)
             {
-                Logger.Error(ex);
-                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.ToString());
+                var errorMessage = "An unexpected error occured trying to get this page localization, consolt the logs for more details.";
```
Comment (david-poindexter) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs:831]:
```suggestion
                var errorMessage = "An unexpected error occurred trying to get this page localization, consult the logs for more details.";
```

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs [hunk 5]
```diff
@@ -712,12 +857,18 @@ public HttpResponseMessage UpdateTabLocalization(DnnPagesRequest request)
             }
             catch (Exception ex)
             {
-                Logger.Error(ex);
-                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.ToString());
+                var errorMessage = "An unexpected error occured trying to update the page localization, please consult the logs for more details.";
```
Comment (david-poindexter) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs:860]:
```suggestion
                var errorMessage = "An unexpected error occurred trying to update the page localization, please consult the logs for more details.";
```

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs [hunk 6]
```diff
@@ -743,12 +894,18 @@ public HttpResponseMessage RestoreModule(int tabModuleId)
             }
             catch (Exception ex)
             {
-                Logger.Error(ex);
-                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.ToString());
+                var errorMessage = "An unexpected error occured while trying to restore the module onto that page.";
```
Comment (david-poindexter) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs:897]:
```suggestion
                var errorMessage = "An unexpected error occurred while trying to restore the module onto that page.";
```

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs [hunk 7]
```diff
@@ -774,16 +931,17 @@ public HttpResponseMessage DeleteModule(int tabModuleId)
             }
             catch (Exception ex)
             {
-                Logger.Error(ex);
-                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.ToString());
+                var errorMessage = "An unexpected error occured while trying to delete the module, consult the logs for more details.";
```
Comment (david-poindexter) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs:934]:
```suggestion
                var errorMessage = "An unexpected error occurred while trying to delete the module, consult the logs for more details.";
```

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs [hunk 8]
```diff
@@ -793,20 +951,24 @@ public HttpResponseMessage GetContentLocalizationEnabled()
...
                     return this.GetForbiddenResponse();
                 }
+
                 return this.Request.CreateResponse(HttpStatusCode.OK, new { Success = true, this.PortalSettings.ContentLocalizationEnabled });
             }
             catch (Exception ex)
             {
-                Logger.Error(ex);
-                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.ToString());
+                var errorMessage = "An unexpected error occured while trying to find if content localization is enabled";
```
Comment (david-poindexter) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs:959]:
```suggestion
                var errorMessage = "An unexpected error occurred while trying to find if content localization is enabled";
```

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs [hunk 9]
```diff
@@ -816,24 +978,32 @@ public HttpResponseMessage GetCachedItemCount(string cacheProvider, int pageId)
...
                     return this.GetForbiddenResponse();
                 }
+
                 return this.Request.CreateResponse(HttpStatusCode.OK, new { Success = true, Count = OutputCachingProvider.Instance(cacheProvider).GetItemCount(pageId) });
             }
             catch (Exception ex)
             {
-                Logger.Error(ex);
-                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.ToString());
+                var errorMessage = "An unexpected error occured trying to get the cached items count, please consult the logs for more details.";
```
Comment (david-poindexter) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs:986]:
```suggestion
                var errorMessage = "An unexpected error occurred trying to get the cached items count, please consult the logs for more details.";
```

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs [hunk 10]
```diff
@@ -843,14 +1013,12 @@ public HttpResponseMessage ClearCache([FromUri] string cacheProvider, [FromUri]
             }
             catch (Exception ex)
             {
-                Logger.Error(ex);
-                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.ToString());
+                var message = "An unexpected error occured while trying to clear the cache for this page, see logs for more details.";
```
Comment (david-poindexter) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/PagesController.cs:1016]:
```suggestion
                var message = "An unexpected error occurred while trying to clear the cache for this page, see logs for more details.";
```

**Comment by valadas** [comment]
Ok, that was strange but all corrections are now done @david-poindexter (gammar-cop) :P If you could re-approve/merge. Thanks for the review :)

**Comment by cklenk** [comment]
Would this issue been related to anything with trying to upload a file to a page if you are using the file page type?

**Comment by valadas** [comment]
This fixed 2 issues, one with a missed update on a previous change regarding file uploads.
The second part of this fix was a wrong validation regarding file extension mappings to mime types.

I just saw the issue you mean, I guesses that part was missed, i guess the guy who did this PR (me) did not even know there was another uploader there 😁 

The fix should be similar to this PR, if you can create a new issue for it and link to this PR, it would be an easy fix using this code as a reference in that other place.

As a workaround in the meantime you could upload the file you want using the site assets and then select the file instead of uploading it in the new page UI.

**Comment by cklenk** [comment]
@valadas okay I can do that! Thank you.

**Pull request merged**